app/scripts/controllers/provider-approval.js

```diff
@@ -31,19 +31,25 @@ class ProviderApprovalController extends SafeEventEmitter {
    *
    * @param {object} opts - opts for the middleware contains the origin for the middleware
    */
-  createMiddleware ({ origin, getSiteMetadata }) {
+  createMiddleware ({ senderUrl, extensionId, getSiteMetadata }) {
     return createAsyncMiddleware(async (req, res, next) => {
       // only handle requestAccounts
       if (req.method !== 'eth_requestAccounts') return next()
       // if already approved or privacy mode disabled, return early
       const isUnlocked = this.keyringController.memStore.getState().isUnlocked
+      const origin = senderUrl.hostname
       if (this.shouldExposeAccounts(origin) && isUnlocked) {
         res.result = [this.preferencesController.getSelectedAddress()]
         return
       }
       // register the provider request
-      const metadata = await getSiteMetadata(origin)
-      this._handleProviderRequest(origin, metadata.name, metadata.icon)
+      const metadata = { hostname: senderUrl.hostname, origin }
+      if (extensionId) {
```

Comment:
When is this ever not the Metamask extensionId?

Reply:
The `extensionId` is only set if the connection request originated from an extension. The requests from the inpage script come from that site's domain, so the extension id isn't set.

Reply:
hmm I don't think that is the case. 
https://github.com/MetaMask/metamask-extension/blob/8dfb0e8154d258d5c20ee4d7fd0c5b9e478478d6/app/scripts/background.js#L392-L396

Even when the remotePort is on a dapp, the sender id is still the MetaMask extension id, which is breaking provider approval for all dapps.

Reply:
Oh! Hmm, that's not good. I'll investigate.

Reply:
It's only visually on the connection approval. After connecting, the `approvedOrigins` get updated correctly and the connection is established to the dapp.

Reply:
Good catch! I also messed up the name of the site title and icon. I'll have a PR up to fix both issues momentarily.

Reply:
This should be addressed here: https://github.com/MetaMask/metamask-extension/pull/7349